components/BillTestimonies/BillTestimonies.js

```diff
@@ -1,21 +1,28 @@
 import React from "react";
-import { testimonies } from "../MockTestimonies"
 import { Table, Container } from 'react-bootstrap'
 import ExpandTestimony from "../ExpandTestimony/ExpandTestimony";
+import { usePublishedTestimonyListing } from "../db";
+
 
 // the word "testimonies": In more general, commonly used, contexts, the plural form will also be testimony.  However, in more specific contexts, the plural form can also be testimonies e.g. in reference to various types of testimonies or a collection of testimonies.
 
+
 const ViewTestimonies = (props) => {
   const bill = props.bill
-  const testimoniesComponent = !testimonies ? "" :
+  const testimoniesResponse = usePublishedTestimonyListing({billId: bill.id})
+
+  const testimonies = testimoniesResponse.status == "loading" ? [] : testimoniesResponse.result
+
+  const testimoniesComponent = testimonies.loading ? "" :
```

Comment:
We can just do `testimoniesCompnent = testimonies.map...` since testimonies will be an empty array if loading (and we won't render anything)

Reply:
yes